src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -301,13 +251,7 @@ public boolean publish(String message, String color) {
 
     @Override
     public boolean publish(String message, JSONArray attachments, String color) {
-        return publish(
-                SlackRequest.builder()
-                        .withMessage(message)
-                        .withAttachments(attachments)
-                        .withColor(color)
-                        .build()
-        );
+        return publish(message, attachments, null, color, null);
```

Comment:
Route through a centralized location to pick up all the ancillary things that we pass to the `SlackMessageRequestBuilder` (e.g. `asUser`, `iconEmoji`).